probe/probe.go

```diff
@@ -360,119 +379,197 @@ type KubeArmorProbeData struct {
-	_, err := getKubeArmorDaemonset(c, o)
-	return err == nil
+func isKubeArmorRunning(c *k8s.Client, o Options) (bool, *Status) {
+	isRunning, DaemonsetStatus := getKubeArmorDaemonset(c, o)
+	return isRunning, DaemonsetStatus
+
+}
 
+// Status data
+type Status struct {
+	Desired   string `json:"desired"`
+	Ready     string `json:"ready"`
+	Available string `json:"available"`
 }
 
-func getKubeArmorDaemonset(c *k8s.Client, o Options) (bool, error) {
+func getKubeArmorDaemonset(c *k8s.Client, o Options) (bool, *Status) {
 	var data [][]string
 	// KubeArmor DaemonSet
 	w, err := c.K8sClientset.AppsV1().DaemonSets(o.Namespace).Get(context.Background(), "kubearmor", metav1.GetOptions{})
 	if err != nil {
-		return false, err
+		return false, nil
 	}
-	color.Green("\nFound KubeArmor running in Kubernetes\n\n")
-	_, err = boldWhite.Printf("Daemonset :\n")
-	if err != nil {
-		color.Red(" Error while printing")
+	if o.Output == "text" {
+		color.Green("\nFound KubeArmor running in Kubernetes\n\n")
+		_, err = boldWhite.Printf("Daemonset :\n")
+		if err != nil {
+			color.Red(" Error while printing")
+		}
 	}
+
 	desired, ready, available := w.Status.DesiredNumberScheduled, w.Status.NumberReady, w.Status.NumberAvailable
 	if desired != ready && desired != available {
+
 		data = append(data, []string{" ", "kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
 		return false, nil
 	}
+	if o.Output == "json" {
+		DaemonSetStatus := Status{
+			Desired:   strconv.Itoa(int(desired)),
+			Ready:     strconv.Itoa(int(ready)),
+			Available: strconv.Itoa(int(available)),
+		}
+		return true, &DaemonSetStatus
+	}
 	data = append(data, []string{" ", "kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
 	renderOutputInTableWithNoBorders(data)
 	return true, nil
 }
-func getKubeArmorDeployments(c *k8s.Client, o Options) {
-	var data [][]string
-	_, err := boldWhite.Printf("Deployments : \n")
-	if err != nil {
-		color.Red(" Error while printing")
+
+func getKubeArmorDeployments(c *k8s.Client, o Options) map[string]*Status {
+
+	if o.Output == "text" {
+		_, err := boldWhite.Printf("Deployments : \n")
+		if err != nil {
+			color.Red(" Error while printing")
+		}
 	}
 	kubearmorDeployments, err := c.K8sClientset.AppsV1().Deployments(o.Namespace).List(context.Background(), metav1.ListOptions{
 		LabelSelector: "kubearmor-app",
 	})
-
 	if err != nil {
-		return
+		return nil
 	}
+
 	if len(kubearmorDeployments.Items) > 0 {
+
+		DeploymentsData := make(map[string]*Status)
+		var data [][]string
 		for _, kubearmorDeploymentItem := range kubearmorDeployments.Items {
 			desired, ready, available := kubearmorDeploymentItem.Status.UpdatedReplicas, kubearmorDeploymentItem.Status.ReadyReplicas, kubearmorDeploymentItem.Status.AvailableReplicas
 			if desired != ready && desired != available {
```

Comment:
why should we have an if else condition if we are continuing for a failed `if`

Reply:
We are not continuing for the failed if, like in the case of `text` output we won't be printing anything, and in the case of `json` output we are returning an empty structure .

Reply:
Sorry, maybe I wasn't clear on the comment. What I meant is we don't need an else in this condition. If the `if` is true it is going to `continue` without executing anything else right?

Reply:
Yes

Reply:
can we remove the else and sanitize the code?